### PR TITLE
Update proxy to ES2020

### PR DESCRIPTION
**Pull request by kisbg**
Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

**Comment by ksh8281** [comment]
I think something is wrong
could we push your branch again?
i can see the yellow light within over 2days

File: tools/test/test262/excludelist.orig.xml [hunk 1]
```diff
@@ -4427,9 +4428,9 @@
     <test id="built-ins/Proxy/create-handler-is-revoked-proxy"><reason>TODO</reason></test>
     <test id="built-ins/Proxy/create-target-is-revoked-function-proxy"><reason>TODO</reason></test>
     <test id="built-ins/Proxy/create-target-is-revoked-proxy"><reason>TODO</reason></test>
-    <test id="built-ins/Proxy/defineProperty/targetdesc-not-configurable-writable-desc-not-writable"><reason>TODO</reason></test>
-    <test id="built-ins/Proxy/deleteProperty/targetdesc-is-configurable-target-is-not-extensible"><reason>TODO</reason></test>
-    <test id="built-ins/Proxy/getOwnPropertyDescriptor/resultdesc-is-not-configurable-not-writable-targetdesc-is-writable"><reason>TODO</reason></test>
+    <test id="built-ins/Proxy/defineProperty/return-boolean-and-define-target"><reason>TODO</reason></test>
+    <test id="built-ins/Proxy/set/trap-is-missing-receiver-multiple-calls"><reason>TODO</reason></test>
+    <test id="built-ins/Proxy/set/trap-is-missing-receiver-multiple-calls-index"><reason>TODO</reason></test>
```
Comment (clover2123) [inline tools/test/test262/excludelist.orig.xml:4433]: You have excluded some TCs (even some test262 cases) for this patch. 
Is there any conflict between old and new standard?
Would you please explain this?

**Comment by kisbg** [comment]
@clover2123  i updated the pull request. Thank you for the review!

**Comment by clover2123** [comment]
@kisbg 
Is there still some conflicts in v8 and spidermonkey TC?

**Comment by kisbg** [comment]
@clover2123  I added a few test-cases in V8 and Spidermonkey skip list, since these test-cases are outdated. 
Also few test262 proxy test-cases remained skipped since  they are failing on other engines as well.

File: tools/test/spidermonkey/excludelist.txt [hunk 2]
```diff
@@ -10,6 +10,9 @@ non262/extensions/regress-367501-03.js
 non262/extensions/regress-367501-04.js
 
 # Skip, These are dependent on implementation or not standard.
+non262/Array/shift-no-has-trap.js
+non262/Array/pop-no-has-trap.js
+non262/Array/concat-proxy.js
```
Comment (clover2123) [inline tools/test/spidermonkey/excludelist.txt:15]: These skipped spidermonkey TCs are still valid in other major browsers (I tested it on chrome and firefox both).
Would you please check it once again?

Comment (kisbg) [reply tools/test/spidermonkey/excludelist.txt]: i fixed the error. The test-cases has been removed from the skip list.

**Pull request merged**